File: src/main/java/com/indiepost/repository/StaticPageRepository.java
package com.indiepost.repository;

import com.indiepost.dto.StaticPageDto;
import com.indiepost.enums.Types;
import com.indiepost.model.StaticPage;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

/**
 * Created by jake on 17. 3. 5.
 */
public interface StaticPageRepository {

    Long save(StaticPage staticPage);

    StaticPage findById(Long id);

    void update(StaticPage staticPage);

    void delete(StaticPage staticPage);

    Page<StaticPageDto> find(Pageable pageable);

    Page<StaticPageDto> find(Pageable pageable, Types.PostStatus pageStatus);

    Long count();

    Long count(Types.PostStatus pageStatus);

    StaticPage findBySlug(String slug);

    void bulkUpdateStatusByIds(List<Long> ids, Types.PostStatus status);

    void bulkDeleteByIds(List<Long> ids);

    void bulkDeleteByStatus(Types.PostStatus status);

    boolean isExists(Long id);
}

File: src/main/kotlin/com/indiepost/repository/StaticPageRepository.kt
package com.indiepost.repository

import com.indiepost.dto.StaticPageDto
import com.indiepost.enums.Types
import com.indiepost.model.StaticPage
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

/**
 * Created by jake on 17. 3. 5.
 */
interface StaticPageRepository {

    fun save(staticPage: StaticPage): Long?

    fun findById(id: Long?): StaticPage

    fun update(staticPage: StaticPage)

    fun delete(staticPage: StaticPage)

    fun find(pageable: Pageable): Page<StaticPageDto>

    fun find(pageable: Pageable, pageStatus: Types.PostStatus): Page<StaticPageDto>

    fun count(): Long

    fun count(pageStatus: Types.PostStatus): Long

    fun findBySlug(slug: String): StaticPage?

    fun bulkUpdateStatusByIds(ids: List<Long>, status: Types.PostStatus)

    fun bulkDeleteByIds(ids: List<Long>)

    fun bulkDeleteByStatus(status: Types.PostStatus)

    fun isExists(id: Long?): Boolean
}
